### PR TITLE
Replaced button, input and field components with web-components

### PR DESCRIPTION
**Pull request by dmitryrazinkov**

### CODE REVIEW

File: src/components/VPassword.vue [hunk 1]
```diff
@@ -1,35 +1,34 @@
...
+    ></wc-svg-icon>
 
-    <template #error>
+    <wc-error slot="error">
       <slot name="error"></slot>
-    </template>
+    </wc-error>
 
-    <template #hint>
+    <wc-hint slot="hint">
```
Comment (Yar-s) [inline src/components/VPassword.vue:15]: Shall we allow the password component control error and hint views?
By doing this we make its usage different from `wc-input` component.

File: src/components/VSearch.vue [hunk 2]
```diff
@@ -1,35 +1,34 @@
...
-    </template>
 
-    <template #error>
+    <wc-error slot="error">
       <slot name="error"></slot>
-    </template>
+    </wc-error>
 
-    <template #hint>
+    <wc-hint slot="hint">
```
Comment (Yar-s) [inline src/components/VSearch.vue:15]: The same question as for the `VPassword` component

File: src/views/CatsView.vue [hunk 3]
```diff
@@ -4,17 +4,15 @@
...
         v-model="search"
-      ></v-input>
+      ></v-search>
 
-      <v-button class="cats-view__add-btn button--icon">
-        <template #icon>
-          <wc-svg-icon href="/icons.svg" symbol="add"></wc-svg-icon>
-        </template>
-      </v-button>
+      <wc-button size="big" class="cats-view__add-btn" iconOnly="true">
```
Comment (Yar-s) [inline src/views/CatsView.vue:13]: Looks like we can pass boolean props without specifying their 'true' values
```suggestion
      <wc-button size="big" class="cats-view__add-btn" iconOnly>
```

File: src/components/LoginForm.vue [hunk 4]
```diff
@@ -9,17 +9,18 @@
...
+    <wc-input
       class="login-form__input"
       label="Email"
       name="email"
+      id="email"
       :invalid="!!emailError"
       v-model="email"
     >
-      <template #error>
+      <wc-error v-if="emailError" slot="error">
```
Comment (Yar-s) [inline src/components/LoginForm.vue:20]: Input shows an error when it's invalid. We don't need to control this.
```suggestion
      <wc-error slot="error">
```

File: src/components/VSearch.vue (hunk 2 above)
Comment (dmitryrazinkov) [reply src/components/VSearch.vue]: fixed

File: src/components/VPassword.vue (hunk 1 above)
Comment (dmitryrazinkov) [reply src/components/VPassword.vue]: done

**Review by Yar-s — APPROVED** [review]

**Pull request merged**